test/e2e/set/scyllacluster/scyllacluster_upgrades.go

```diff
@@ -109,6 +108,16 @@ var _ = g.Describe("ScyllaCluster upgrades", func() {
 				o.Expect(err).NotTo(o.HaveOccurred())
 			}
 			scyllaclusterverification.VerifyCQLData(ctx, di)
+
+			scyllaClient, hosts, err := utils.GetScyllaClient(ctx, f.KubeClient().CoreV1(), sc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			for _, host := range hosts {
+				snapshots, err := scyllaClient.ListSnapshots(ctx, host)
```

Comment:
For comprehensibility 1 year from now, I suggest adding a comment explaining the intention - snapshots AFAICT aren't synonymous with post upgrade hooks, especially given the possibility of giving up snapshots during upgrade in the future altogether.

For example:

```
// Ensure that the post upgrade hooks have been run on every single node
```